JanggiGame.py

```diff
@@ -63,6 +63,17 @@ def get_board(self):
         """getter for board"""
         return self._board
 
+    def generate_piece_objects(self):
+        """generator yields all the piece objects from the game board"""
+        for row in self.get_board():
+            for piece_obj in row:
+                if piece_obj is not None:
+                    yield piece_obj
+
+    def get_all_pieces(self):
+        """returns a list of the current piece objects"""
+        return list(self.generate_piece_objects())
```

Comment:
Keep in mind, this method will always create and return a full list. But calling the generator will return one item at a time, which could be faster overall (depending on the use-case).